solana/bridge-escrow/programs/bridge-escrow/src/lib.rs

```diff
@@ -83,16 +99,39 @@ pub mod bridge_escrow {
 
         // save intent on a PDA derived from the auctioneer account
         let intent = &mut ctx.accounts.intent;
-        intent.intent_id = intent_id;
+
+        let current_timestamp = Clock::get()?.unix_timestamp as u64;
+
+        intent.intent_id = intent_id.clone();
         intent.user = user_in;
         intent.token_in = token_in;
         intent.amount_in = amount_in;
-        intent.token_out = token_out;
+        intent.token_out = token_out.clone();
         intent.timeout_timestamp_in_sec = timeout_in_sec;
```

Comment:
Not exactly part of this PR, but is the timeout relative duration or is it a timestamp? This is also what I’ve commented on the other PR.

Reply:
resolved in the other PR.